glib/src/param_spec.rs

```diff
@@ -377,96 +377,66 @@ macro_rules! define_param_spec_min_max {
 }
 
 macro_rules! define_param_spec_numeric {
-    ($rust_type:ident, $ffi_type:path, $value_type:ty, $rust_type_offset:expr, $from_glib:expr) => {
+    ($rust_type:ident, $ffi_type:path, $value_type:ty, $rust_type_offset:expr, $ffi_fun:ident, $alias:literal) => {
         define_param_spec!($rust_type, $ffi_type, $rust_type_offset);
-        define_param_spec_default!($rust_type, $value_type, $from_glib);
-        define_param_spec_min_max!($rust_type, $value_type, $from_glib);
+        define_param_spec_default!($rust_type, $value_type, |x| x);
+        define_param_spec_min_max!($rust_type, $value_type, |x| x);
+
+        impl $rust_type {
+            #[allow(clippy::new_ret_no_self)]
+            #[doc(alias = $alias)]
+            pub fn new(
+                name: &str,
+                nick: &str,
+                blurb: &str,
+                minimum: $value_type,
+                maximum: $value_type,
+                default_value: $value_type,
+                flags: ParamFlags,
+            ) -> ParamSpec {
+                assert_param_name(name);
+                unsafe {
+                    from_glib_none(gobject_ffi::$ffi_fun(
+                        name.to_glib_none().0,
+                        nick.to_glib_none().0,
+                        blurb.to_glib_none().0,
+                        minimum,
+                        maximum,
+                        default_value,
+                        flags.into_glib(),
+                    ))
+                }
+            }
+        }
     };
 }
 
+fn assert_param_name(name: &str) {
```

Comment:
This needs to be a macro or use `#[track_caller]`, otherwise the panic messages are going to point at this very line instead of where things actually failed.